fix iruby backend broken in Docker container #2644

kpaharelau opened the pull request:
iruby doesn't work without registered IRuby kernel in the '~/.ipython/kernels/ruby/kernel.json'. This file can be created by command 'iruby register'

kpaharelau committed:
fix iruby backend broken in Docker container #2644

iruby doesn't work without registered IRuby kernel in the '~/.ipython/kernels/ruby/kernel.json'. This file can be created by command 'iruby register'

scottdraves commented on Dockerfile:
```diff
@@ -39,6 +39,8 @@ RUN su -m beaker -c "cd /home/beaker/src  && gradle build"
 RUN su -m beaker -c "julia --eval 'Pkg.add(\"IJulia\")'" && \
     su -m beaker -c "julia --eval 'Pkg.add(\"Gadfly\")'"
 
+RUN su -m beaker -c "iruby register"
```
This should go in the base_docker_image/Dockerfile

kpaharelau committed:
fix iruby backend broken in Docker container #2644

I moved to the base_docker_image/Dockerfile

###################
#      Setup      #
###################

RUN su -m beaker -c "julia --eval 'Pkg.add(\"IJulia\")'" && \
   su -m beaker -c "julia --eval 'Pkg.add(\"Gadfly\")'"

RUN su -m beaker -c "iruby register"

But I do not see these changes in fs  when I run 'sudo docker build -t ubuntu/beaker . '

All works fine when I use this code in the root Dockerfile

I think that all will be works when https://hub.docker.com/r/beakernotebook/beaker/ will be uses branch with my changes (edite

scottdraves commented on Dockerfile:
```diff
@@ -36,10 +36,7 @@ RUN su -m beaker -c "cd /home/beaker/src  && gradle build"
-    su -m beaker -c "julia --eval 'Pkg.add(\"Gadfly\")'"
-
-RUN mkdir -p /home/beaker/.beaker/v1/config && \
+WhebRUN mkdir -p /home/beaker/.beaker/v1/config && \
```
what is this? "Wheb"

kpaharelau committed:
fix iruby backend broken in Docker container #2644

please do not change julia, just fix iruby.
if there is a problem with julia, please file an issue. FIXED

But I have little question

I see same code for Julia in both dockerfiles

base_docker_image/Dockerfile

RUN apt-get install -y julia && \
    julia --eval 'Pkg.add("IJulia")' && \
    julia --eval 'Pkg.add("Gadfly")'

Dockerfile

RUN su -m beaker -c "julia --eval 'Pkg.add(\"IJulia\")'" && \
    su -m beaker -c "julia --eval 'Pkg.add(\"Gadfly\")'"

For ruby init code must be only in base_docker_image/Dockerfile

RUN su -m beaker -c "iruby register"

I do not quite understand what ruby is different from julia
